tests_common/test_utils/compat.py

```diff
@@ -44,6 +45,11 @@
 AIRFLOW_V_2_10_PLUS = Version(AIRFLOW_VERSION.base_version) >= Version("2.10.0")
 AIRFLOW_V_3_0_PLUS = Version(AIRFLOW_VERSION.base_version) >= Version("3.0.0")
 
+if AIRFLOW_V_3_0_PLUS:
```

Comment:
I do not fully get it... I'd understand that AIP-44 was not working in Airflow 2.8 and 2.9... and we know we want to remove it soon before 3.0. But in 2.10 at least the tests for AIP-44 should be working...

Why not testing in `if AIRFLOW_V_2_10_PLUS:`?

Reply:
As far as I know, AIP-44 is not yet enabled in Airflow 2.10. If anyone develops a new feature using AIP-44 functionality in providers, it may pass all tests in CI but could break once released because AIP-44 isn't enabled on users' systems. 

For example, the `use_airflow_context` feature won’t work on users' setups since it requires AIP-44. If we release it claiming compatibility with 2.10, it may appear functional in our CI if AIP-44 is enabled, but in reality, it will only work when AIP-44 is activated on the user’s side.

Reply:
@jscheffl just checking, are you with this?